### ISSUE TITLE
[LX] busybox Segmentation Fault

### ISSUE THREAD

**Issue by vizanto** [joyent/smartos-live#526]
```
# /nix/store/ldj18wicxhjf0739kr08p0rbnsby7zr9-busybox
BusyBox v1.22.1 () multi-call binary.
BusyBox is copyrighted by many authors between 1998-2012.
Licensed under GPLv2. See source distribution for detailed
copyright notices.

Usage: busybox [function [arguments]...]
   or: busybox --list
   or: function [arguments]...

        BusyBox is a multi-call binary that combines many common Unix
        utilities into a single executable.  Most people will create a
        link to busybox for each function they wish to use and BusyBox
        will act like whatever it was invoked as.

Currently defined functions:
        ash, mkdir, tar, unxz, xzcat

Segmentation fault (core dumped)
```

Binary and core dump, here:  http://vizanto.com/busysegv/busysegv.zip

**Comment by jjelinek** [comment]
This is failing because the application is attempting to execute code on the stack. From the system's console:
2015-12-16T14:04:42.575380+00:00 headnode genunix: [ID 533030 kern.notice] NOTICE: ldj18wicxhjf0739[136152] attempt to execute code on stack by uid 1000

We explicitly run with a non-executable stack for security reasons.

**Comment by vizanto** [comment]
How can I check if the same happens on NixOS linux?

**Comment by jjelinek** [comment]
You would have to install and test it there, but that wouldn't mean anything for running inside an lx zone on SmartOS, since each OS can choose the level of security they want to provide, so NixOS might not be concerned about an executable stack, and thus allow it.

**Comment by pfmooney** [comment]
@vizanto I recently found a bug in the way our emulation was setting initial register state when beginning the execution of certain binaries.  (Static binaries, particularly those compiled with uClibc, seemed to be the most harshly effected.)  I tested the busybox binary you provided on a PI with the fix and did not experience any SIGSEGVs.

If you happen to test on a platform with the fix (742ed6585f2871e1be923295a58212ff65bfe9d3), I would appreciate it if you would report back with your findings.
